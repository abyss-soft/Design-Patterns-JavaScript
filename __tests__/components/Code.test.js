import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import themeLight from '../../src/styles/hljs/hljs.light';
import themeDark from '../../src/styles/hljs/hljs.dark';
import Code from '../../src/components/Code';

const mockStore = configureMockStore();
const initialStore = {
  progress: {
    current: {
      uuid: '5',
      answered: false,
      correct: null,
      codeES5: 'JavaScript ES5',
      codeES6: 'JavaScript ES6'
    }
  },
  js: 'es5'
};
const storeES5 = mockStore(initialStore);
const storeES6 = mockStore({ ...initialStore, js: 'es6' });

describe('<Code /> component', () => {
  it('renders ES5 code in LIGHT mode', () => {
    const tree = renderer
      .create(
        <Provider store={storeES5}>
          <Code style={themeLight} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ES6 code in DARK mode', () => {
    const tree = renderer
      .create(
        <Provider store={storeES6}>
          <Code style={themeDark} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
