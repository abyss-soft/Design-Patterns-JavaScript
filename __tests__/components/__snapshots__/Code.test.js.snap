// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> component renders ES5 code in LIGHT mode 1`] = `
<pre
  className="fixed"
  style={
    Object {
      "background": "#fafafa",
      "border": "1px solid #d8d8d8",
      "color": "#383a42",
      "display": "block",
      "overflowX": "auto",
      "padding": "1em",
    }
  }
>
  <code>
    JavaScript ES5
  </code>
</pre>
`;

exports[`<Code /> component renders ES6 code in DARK mode 1`] = `
<pre
  className="fixed"
  style={
    Object {
      "background": "#282828",
      "border": "1px solid #555",
      "color": "#ebdbb2",
      "display": "block",
      "overflowX": "auto",
      "padding": "1em",
    }
  }
>
  <code>
    JavaScript ES6
  </code>
</pre>
`;
